markdown/templates.go

```diff
@@ -0,0 +1,85 @@
+package markdown
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+)
+
+// TemplatesDir is the directory where the markdown templates are stored
+const TemplatesDir = "./templates"
+
+// The template for shared app content
+var AppSharedTemplate *template.Template
+
+// LoadAppTemplate loads the content for a specific app
+func LoadAppTemplate(name string) (*template.Template, error) {
+	filePath := fmt.Sprintf("apps/%s", name)
+	tpl, err := loadTemplate(filePath)
+	if err != nil {
+		return nil, err
+	}
+	// add the shared template
+	tpl, err = tpl.ParseFiles(fmt.Sprintf("%s/apps/shared.md", TemplatesDir))
+
+	return tpl, err
+}
+
+// LoadAllTemplates loads all templates from the templates directory
+func LoadAllTemplates() error {
```

Comment:
Could this return the shared template instead so we don't need to use a package global variable?

Reply:
The templates are loaded one time on app start, then the `execute` function uses this variable to merge the models later on during the export process. I could add it to the `App` struct which would make it available to all of the child models. Would it make sense to have it there?

Reply:
I guess it doesn't matter too much in a CLI application where there will only ever be one instance per process. In `lnd` we try to avoid using package global variables as they could be accessed by multiple goroutines and lead to weird side effects. But yes, in general it's good practice to keep state in structs and pass those around if necessary.